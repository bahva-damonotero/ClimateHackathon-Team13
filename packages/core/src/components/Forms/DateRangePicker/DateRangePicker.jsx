import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import dateRangePicker from '@uswds/uswds/js/usa-date-range-picker';

import { createClassString } from '@boozallen-uip/uswds-react-utils';

export default function DateRangePicker({ children, className }) {
  const containerRef = useRef();

  useEffect(() => {
    const containerEl = containerRef.current;
    if (!containerEl) return;

    // call USWDS's `on` function to initialize the component
    dateRangePicker.on(containerEl);

    // on dismount, call USWDS's `off` function to cleanup the component
    return () => {
      dateRangePicker.off(containerEl);
    };
  }, []);

  return (
    <div
      className={createClassString(['bah-uswds-date-range-picker', className])}
      ref={containerRef}
    >
      <div className="usa-date-range-picker">{children}</div>
    </div>
  );
}

DateRangePicker.propTypes = {
  /** Should contain two `<DatePicker />` node(s) */
  children: PropTypes.node.isRequired,
  /** Any additional classes to apply */
  className: PropTypes.string
};
