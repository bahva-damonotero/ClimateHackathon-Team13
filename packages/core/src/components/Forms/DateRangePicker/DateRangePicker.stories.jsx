/* eslint-disable react/prop-types */
import React from 'react';

import DateRangePicker from '.';
import { DatePicker, Form } from '..';

export default {
  title: 'Forms/DateRangePicker',
  component: DateRangePicker,
  subcomponents: {
    DatePicker
  },
  argTypes: {
    startLabel: {
      name: 'start <DatePicker> children',
      description: 'Label for the start `<DatePicker>`',
      type: { required: true },
      table: {
        type: { summary: 'node' }
      }
    },
    startId: {
      name: 'start <DatePicker> id',
      description: 'ID for the start `<DatePicker>`',
      type: { required: true },
      table: {
        type: { summary: 'string' }
      }
    },
    endLabel: {
      name: 'end <DatePicker> children',
      description: 'Label for the end `<DatePicker>`',
      type: { required: true },
      table: {
        type: { summary: 'node' }
      }
    },
    endId: {
      name: 'end <DatePicker> id',
      description: 'ID for the end `<DatePicker>`',
      type: { required: true },
      table: {
        type: { summary: 'string' }
      }
    },
    children: {
      control: { type: null }
    }
  }
};

const Template = ({
  startId,
  startLabel,
  endId,
  endLabel,
  disabled,
  ...args
}) => (
  <Form>
    <DateRangePicker {...args}>
      <DatePicker disabled={disabled} id={startId}>
        {startLabel}
      </DatePicker>

      <DatePicker disabled={disabled} id={endId}>
        {endLabel}
      </DatePicker>
    </DateRangePicker>
  </Form>
);

const defaultArgs = {
  startId: 'date-range-picker-example-start',
  endId: 'date-range-picker-example-end',
  startLabel: 'Start date',
  endLabel: 'End date'
};

export const def = Template.bind({});
def.storyName = 'Default';
def.args = {
  ...defaultArgs
};
