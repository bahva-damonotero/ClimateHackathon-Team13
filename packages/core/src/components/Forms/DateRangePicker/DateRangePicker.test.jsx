import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import dateRangePicker from '@uswds/uswds/js/usa-date-range-picker';

import DateRangePicker from '.';
import { DatePicker } from '..';

describe('DateRangePicker', () => {
  const jsx = (
    <DateRangePicker>
      <DatePicker id="date-range-picker-test-start">
        Start Date Label
      </DatePicker>

      <DatePicker id="date-range-picker-test-end">End Date Label</DatePicker>
    </DateRangePicker>
  );

  it('should call `dateRangePicker.on()` to initialize the component', () => {
    const onSpy = jest.spyOn(dateRangePicker, 'on');

    render(jsx);

    expect(onSpy).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(jsx).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
