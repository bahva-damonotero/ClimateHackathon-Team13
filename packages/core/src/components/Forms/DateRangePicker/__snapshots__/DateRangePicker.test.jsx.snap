// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker should match snapshot 1`] = `
<div
  className="bah-uswds-date-range-picker"
>
  <div
    className="usa-date-range-picker"
  >
    <div
      className="bah-uswds-date-picker"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <label
        className="usa-label"
        htmlFor="date-range-picker-test-start"
        id="date-range-picker-test-start-label"
      >
        Start Date Label
      </label>
      <span
        className="usa-hint"
        id="date-range-picker-test-start-hint"
      >
        mm/dd/yyyy
      </span>
      <div
        className="usa-date-picker"
      >
        <input
          aria-describedby="date-range-picker-test-start-hint"
          aria-labelledby="date-range-picker-test-start-label"
          className="usa-input"
          disabled={false}
          id="date-range-picker-test-start"
          name="date-range-picker-test-start"
          type="text"
        />
      </div>
    </div>
    <div
      className="bah-uswds-date-picker"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <label
        className="usa-label"
        htmlFor="date-range-picker-test-end"
        id="date-range-picker-test-end-label"
      >
        End Date Label
      </label>
      <span
        className="usa-hint"
        id="date-range-picker-test-end-hint"
      >
        mm/dd/yyyy
      </span>
      <div
        className="usa-date-picker"
      >
        <input
          aria-describedby="date-range-picker-test-end-hint"
          aria-labelledby="date-range-picker-test-end-label"
          className="usa-input"
          disabled={false}
          id="date-range-picker-test-end"
          name="date-range-picker-test-end"
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
